src/v/cloud_storage/recursive_directory_walker.cc

```diff
@@ -48,10 +52,13 @@ struct walk_accumulator {
                   .value_or(file_stats.time_accessed);
 
             current_cache_size += static_cast<uint64_t>(file_stats.size);
-            files.push_back(
-              {last_access_timepoint,
-               (std::filesystem::path(target) / entry.name.data()).native(),
-               static_cast<uint64_t>(file_stats.size)});
+
+            if (filter && filter.value()(target)) {
```

Comment:
If no filter is provided, this will match no files -- I think we probably want the opposite behavior?

Reply:
Yeah, fat fingered it. Fixed.